clippy_lints/src/explicit_write.rs

```diff
@@ -80,13 +81,8 @@ impl<'tcx> LateLintPass<'tcx> for ExplicitWrite {
                     )
                 };
                 let msg = format!("use of `{}.unwrap()`", used);
-                if let [write_output] = *format_args.format_string_parts {
-                    let mut write_output = write_output.to_string();
-                    if write_output.ends_with('\n') {
-                        write_output.pop();
-                    }
-
-                    let sugg = format!("{}{}!(\"{}\")", prefix, sugg_mac, write_output.escape_default());
+                if let Some(inputs_snippet) = snippet_opt(cx, format_args.inputs_span()) {
+                    let sugg = format!("{}{}!({})", prefix, sugg_mac, inputs_snippet);
```

Comment:
I think you can use `snippet_with_applicability` here and remove the else `span_lint_and_sugg` case. Otherwise this looks good!